### ISSUE TITLE
undefined method `keys' for "Endpoint request timed out":String

### ISSUE THREAD

**Issue by weppos** [DripEmail/drip-ruby#56]
This is a follow up from https://github.com/DripEmail/drip-ruby/issues/33#issuecomment-394511866 as requested.

The issue is still not fixed. Here's the stacktrace:

```
NoMethodError · undefined method `keys' for "Endpoint request timed out":String

gems/drip-ruby-3.2.0/lib/drip/resource.rb:32:in `block in process': undefined method `keys' for "Endpoint request timed out":String (NoMethodError)
    from gems/drip-ruby-3.2.0/lib/drip/resource.rb:31:in `tap'
    from gems/drip-ruby-3.2.0/lib/drip/resource.rb:31:in `process'
    from gems/drip-ruby-3.2.0/lib/drip/resource.rb:9:in `initialize'
    from gems/drip-ruby-3.2.0/lib/drip/response.rb:59:in `new'
    from gems/drip-ruby-3.2.0/lib/drip/response.rb:59:in `block (2 levels) in parse_members'
    from gems/drip-ruby-3.2.0/lib/drip/response.rb:52:in `each'
    from gems/drip-ruby-3.2.0/lib/drip/response.rb:52:in `block in parse_members'
    from gems/drip-ruby-3.2.0/lib/drip/response.rb:50:in `tap'
    from gems/drip-ruby-3.2.0/lib/drip/response.rb:50:in `parse_members'
    from gems/drip-ruby-3.2.0/lib/drip/response.rb:13:in `initialize'
    from gems/drip-ruby-3.2.0/lib/drip/client.rb:119:in `new'
    from gems/drip-ruby-3.2.0/lib/drip/client.rb:119:in `build_response'
    from gems/drip-ruby-3.2.0/lib/drip/client.rb:85:in `make_request'
    from gems/drip-ruby-3.2.0/lib/drip/client.rb:65:in `post'
    from gems/drip-ruby-3.2.0/lib/drip/client/subscribers.rb:54:in `create_or_update_subscriber'
```

It looks like your API are returning data in a format your client is unable to process.

**Comment by weppos** [comment]
It looks like the client is trying to parse the response with any HTTP response type except 429.

https://github.com/DripEmail/drip-ruby/blob/a2b01fda0e6f294d30d7b14567d29a3027dc7089/lib/drip/response.rb#L46-L64

However, it's likely your server is returning something that is neither a 429, neither a valid response. It seems it is returning a body with only the String

> "Endpoint request timed out"

We can't provide more details as we are unable to properly log the request via the client (as #33) was pointing out.

Long story short, there is a bug where either the client is unable to handle the error types your API are serving, or the API are serving responses that are not supposed to be served.

**Comment by bfortune347** [comment]
Hmm, thanks for reporting this. 

We're in the middle of several priorities at the moment (got split testing for workflows coming out, product back-in-stock trigger functionality, and ramping up on date-based workflow triggers), but will log this one for the future. 

-Brendan Fortune
Sr. Director of Product @ Drip

**Comment by jclusso** [comment]
@bfortune347 it's been 3+ years since this was opened in #54. Are you guys going to fix it?

**Comment by mnadel** [comment]
@will-in-wi Mind taking a look at this? Thanks!

**Comment by jclusso** [comment]
@neme101 we had this issue come up 12 days ago. Our integration isn't heavily used, but unless something has changed we still have this issue.

**Comment by jclusso** [comment]
@neme101 I don't believe we were at the time of the error, but we definitly are now. I'll let you know if we see it again.
